pkg/addonfactory/addondeploymentconfig.go

```diff
@@ -109,6 +119,47 @@ func ToAddOnProxyConfigValues(config addonapiv1alpha1.AddOnDeploymentConfig) (Va
 	return values, nil
 }
 
+// ToAddOnResourceRequirementsValues only transform the AddOnDeploymentConfig ResourceRequirements part into Values object that has
+// a specific for helm chart values
+// for example: the spec of one AddOnDeploymentConfig is:
+//
+//	{
+//	  resourceRequirements:[{containerID: "*:*:*", resources: { limits: { memory: "4Gi"}, requests: { memory: "512Mi"}}}],
+//	}
+//
+// after transformed, the Values will be:
+// map[global:map[resourceRequirements:[map[containerIDRegExp:"^.+:.+:.+$" resources:map[limits:map[memory:4Gi]
+// requests:map[memory:512Mi]]]]]]
+func ToAddOnResourceRequirementsValues(config addonapiv1alpha1.AddOnDeploymentConfig) (Values, error) {
+	if config.Spec.ResourceRequirements == nil {
+		return nil, nil
+	}
+
+	resourceRequirements, err := getRegexResourceRequirements(config.Spec.ResourceRequirements)
+	if err != nil {
+		return nil, err
+	}
+
+	type global struct {
+		ResourceRequirements []regexResourceRequirements `json:"resourceRequirements"`
```

Comment:
does this need to be implemented for all addons?

Reply:
The answer is Yes for all helm chart based add-ons that want to support the resource requirements settings.

Reply:
@zhiweiyin318 what is the best place to doc this? it seems to me this is hidden in the code and hard for the user to understand.

Reply:
I think we can add it to the chapter `Values definition` here https://open-cluster-management.io/docs/developer-guides/addon/ .